Text modeling eval

LeonGuertler opened the pull request.

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
testing

LeonGuertler committed:
fix multi-gpu

LeonGuertler committed:
fix multi-gpu

LeonGuertler committed:
fix multi-gpu

LeonGuertler committed:
fix multi-gpu

LeonGuertler committed:
added alternative tokenizers

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
added byte level accuracy and byte-level perplexity

LeonGuertler committed:
fixed byte ppl

LeonGuertler committed:
fixed byte ppl

LeonGuertler committed:
added faster bpe (maybe buggy)

LeonGuertler committed:
added subsample tokenizer

LeonGuertler committed:
added subsample tokenizer

LeonGuertler committed:
added subsample tokenizer

LeonGuertler committed:
added subsample tokenizer

LeonGuertler committed:
added subsample tokenizer

LeonGuertler committed:
added subsample tokenizer

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
added subsample tokenizer(debugging)

LeonGuertler committed:
should be working now

LeonGuertler committed:
added wide config

LeonGuertler committed:
added new tokenizer type

LeonGuertler committed:
added new tokenizer type (debugging)

LeonGuertler committed:
added new tokenizer type (debugging)

LeonGuertler committed:
added new tokenizer type (debugging)

LeonGuertler committed:
added new tokenizer type (debugging)

LeonGuertler committed:
added new tokenizer type (debugging)

LeonGuertler committed:
added new tokenizer type (debugging)

LeonGuertler committed:
added c-proj sharing

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
added bigger runs

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
changed LoRA implementation and added MoE LoRA

LeonGuertler committed:
changed LoRA implementation and added MoE LoRA

LeonGuertler committed:
changed LoRA implementation and added MoE LoRA

LeonGuertler committed:
updated

LeonGuertler committed:
added MoE sharing

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
added MoE sharing (debugging)

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
speed-up

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
added new datasets

LeonGuertler committed:
updated blend of experts model to global

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
more data

LeonGuertler committed:
new weight init for lora

LeonGuertler committed:
rewrite

LeonGuertler committed:
rewrite (debugging)

LeonGuertler committed:
rewrite (debugging)

LeonGuertler committed:
rewrite (debugging)

LeonGuertler committed:
rewrite (debugging)

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
tokenizer -filter to simplify

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

DylanASHillier reviewed (changes requested):
no killing tests tyvm

DylanASHillier commented on configs/generator/standard.yaml:
```diff
@@ -0,0 +1,8 @@
+temperature: 1.2
+top_k: 200
+max_new_tokens: 100
```
[] updating this ting

DylanASHillier commented on configs/train/baseline-large.yaml:
```diff
@@ -0,0 +1,100 @@
+model:
+  core_model_type: generic
+  num_layers: 8
```
maybe change name to baseline_10 and baseline_50

DylanASHillier commented on configs/train/baseline.yaml~:
```diff
@@ -0,0 +1,99 @@
+  num_layers: 8
+  ffn:
+    ffn_type: generic
+    ffn_dim: 1536
```
rmv

DylanASHillier commented on eval.py:
i prefer this to be `model, _ = `

DylanASHillier commented on evals/load_evaluators.py:
```diff
@@ -8,11 +8,12 @@
 from evals.finetuning.qa import FinetuningQA
 from evals.mcqs.benchmarks.stories_progression import ProgressionEvaluator
 
+
```
[] (NOT DONE YET)

DylanASHillier commented on evals/mcqs/load_benchmarks.py:
```diff
@@ -1,24 +1,317 @@
+    """
+    return np.random.choice(
+        dataset_length,
+        dataset_length if num_samples is None else min(num_samples, dataset_length),
```
seed pls

DylanASHillier commented on evals/mcqs/load_benchmarks.py:
```diff
@@ -1,24 +1,317 @@
+        dataset_length if num_samples is None else min(num_samples, dataset_length),
+        replace=False
+    )
+
```
conditional happy on splitting out training eval benchmarks vs testtime eval benchmarks. "don't run popular ones during training"-Leon (so true - Dylan)

DylanASHillier commented on models/build_models.py:
```diff
@@ -15,8 +16,17 @@
 
+from models.experimental.weight_sharing import (
+    SharedInteriorFFNLoraAndCProj,
+    SharedInteriorFFNLora,
```
kill this and use my version :)

DylanASHillier commented on models/build_models.py:
```diff
@@ -30,18 +40,24 @@ def build_model(model_cfg=None, checkpoint=None):
+        current_iter = 0
 
-    return model
+    return model, current_iter
```
training state config instead

DylanASHillier commented on models/build_models.py:
```diff
@@ -151,11 +188,6 @@ def initialize_model(model_cfg):
-    if model_cfg["embedding_weight_tying"]:
-        # share the weights between the token embeddings and the final
-        # logit layer, following: https://paperswithcode.com/method/weight-tying
-        embedding_model.token_embedder.weight = model_head.linear.weight
```
d'accord

DylanASHillier commented on models/components/layers/transformer_blocks.py:
```diff
@@ -36,7 +36,7 @@ def __init__(self, hidden_dim, context_window, use_rope, ffn_cfg, attn_cfg):
         # build the ffn norm
         self.ffn_norm = build_normalization(
-            normalization_name=ffn_cfg["normalization"],
+            normalization_name=ffn_cfg.get("normalization", "rms_norm"), # Default: rms_norm
```
don't like

DylanASHillier commented on models/experimental/hugging_face.py:
```diff
@@ -28,13 +28,13 @@ def build_model(model_cfg):
         trust_remote_code=True,
         attn_implementation=attn_impl,
-        torch_dtype=torch.float16,
+        torch_dtype=torch.float32,
```
???? try understand me no get

DylanASHillier commented on models/experimental/hugging_face.py:
```diff
@@ -129,10 +129,11 @@ def __init__(self, model_cfg):
-        print("Note: Freezing the parameters of the hf_core model.")
-        for param in self.model.parameters():
-            param.requires_grad = False
+        if model_cfg.get("freeze", True):
```
sure

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
debugging

LeonGuertler committed:
training is working

LeonGuertler committed:
added byte-level metrics

LeonGuertler committed:
updated config

LeonGuertler committed:
updated config

LeonGuertler committed:
updated config

LeonGuertler committed:
updated config

LeonGuertler committed:
updated config

LeonGuertler committed:
fixed all evals and WandB

bobbycxy committed:
Added bytelevel() to pre-tokenizer and decoder of tokenizer in BPE

Adjusted the shape of the logits.

LeonGuertler committed:
minor changes

LeonGuertler committed:
minor changes

LeonGuertler committed:
minor changes

LeonGuertler committed:
changed baseline setup

LeonGuertler committed:
changed baseline setup

bobbycxy committed:
Merge branch 'text-modeling-eval' of https://github.com/LeonGuertler/SuperTinyLanguageModels into text-modeling-eval

bobbycxy committed:
added max new tokens for standard.yaml

replaced the static generator class usage with one that uses a generator_dict

troubleshooted the issues where the beam_search flattened the results, when the embedding_model object decodes as a batch.

LeonGuertler committed:
minor fixes

LeonGuertler committed:
.

bobbycxy committed:
added entropy based temperature sampling

revised the standardgenerator to work with single and batched logits

LeonGuertler committed:
revert to rndm idx in dataset

LeonGuertler committed:
fixes

LeonGuertler committed:
adjust hp

LeonGuertler committed:
hp

LeonGuertler committed:
hp

LeonGuertler committed:
added baseline configs

LeonGuertler committed:
temporarily removed tokenizers

bobbycxy committed:
adding 10B T for Fineweb-edu

bobbycxy committed:
revise max and warmup iters

bobbycxy committed:
revised the max and warmup iters, and switched to 10B dataset.

DylanASHillier commented on models/embedding_models.py:
```diff
@@ -115,6 +117,30 @@ def __init__(self, model_cfg):
+
+        self.token_byte_length_cache = self._precompute_byte_lengths()
+
+    def _precompute_byte_lengths(self):
```
why was this added to the interface?

DylanASHillier commented on trainers/scheduler.py:
```diff
@@ -45,87 +45,3 @@ def get_lr(self, iter_num):
-        super().__init__(start_dropout_p)
-        self.start_iter = start_iter
-        self.end_iter = end_iter
-        self.start_dropout_p = start_dropout_p
```
??? why remove linear one is still useful?

LeonGuertler committed:
.item()

LeonGuertler committed:
wip

LeonGuertler committed:
wip

LeonGuertler committed:
wip

LeonGuertler committed:
wip

LeonGuertler committed:
wip

LeonGuertler committed:
wip

bobbycxy committed:
added adaptive sampler with varentropy. Note the addition of attention_type in config

DylanASHillier committed:
updates huggingface code to build but not quite wokring

DylanASHillier committed:
updated loralinear to fix

LeonGuertler committed:
wip

LeonGuertler committed:
wip

LeonGuertler committed:
wip

LeonGuertler committed:
wip

DylanASHillier committed:
partially fixes

DylanASHillier committed:
Merge branch 'text-modeling-eval' of https://github.com/LeonGuertler/SuperTinyLanguageModels into text-modeling-eval

DylanASHillier committed:
adds llama config

bobbycxy committed:
data generation strategy
